packages/hw-app-xtz/src/Tezos.js

```diff
@@ -152,7 +152,43 @@ export default class Tezos {
       }
       response = await this.transport.send(0x80, apdu, code, curve, data);
     }
-    const signature = response.slice(0, response.length - 2).toString("hex");
+    let signature
+    if(curve == 0) {
+      // tz1 signatures come correctly formatted from the ledger.
+      signature = response.slice(0, response.length - 2).toString("hex");
+    } else {
+      // tz2 and tz3 signatures are in DER format from the ledger.
+      const parseError = function() { throw new Error("Cannot parse ledger response."); }
+      var signature_buffer = new Buffer(64);
+      signature_buffer.fill(0);
+      var r_val = signature_buffer.subarray(0, 32);
+      var s_val = signature_buffer.subarray(32,64);
+      var idx=0;
+      const frameType=response.readUInt8(idx++);
+      // Accept either an ASN.1 sequence or ASN.1 set
+      if(frameType != 0x31 && frameType != 0x30) parseError();
+
+      if(response.readUInt8(idx++)+4 != response.length) parseError();
```

Comment:
why the 4 offset? I just see one readUInt8 above.

Reply:
Because of the extra two bytes at the end of the overall ledger response; I can make this clearer.

Reply:
The length also doesn't include the length byte itself.